server/graphql/core/src/loader/asset_status.rs

```diff
@@ -0,0 +1,39 @@
+use repository::asset_log::{AssetLogFilter, AssetLogRepository, AssetLogSort, AssetLogSortField};
+use repository::asset_log_row::Status;
+use repository::{EqualFilter, Pagination};
+use repository::{RepositoryError, StorageConnectionManager};
+
+use async_graphql::dataloader::*;
+use async_graphql::*;
+use std::collections::HashMap;
+
+pub struct AssetStatusLoader {
+    pub connection_manager: StorageConnectionManager,
+}
+
+#[async_trait::async_trait]
+impl Loader<String> for AssetStatusLoader {
+    type Value = Status;
+    type Error = RepositoryError;
+
+    async fn load(&self, ids: &[String]) -> Result<HashMap<String, Self::Value>, Self::Error> {
+        let connection = self.connection_manager.connection()?;
+        let repo = AssetLogRepository::new(&connection);
+
+        let filter = AssetLogFilter::new().asset_id(EqualFilter::equal_any(ids.to_owned()));
+
+        let sort = AssetLogSort {
+            key: AssetLogSortField::LogDatetime,
+            desc: Some(false),
+        };
+
+        let result = repo

```

Comment:
perhaps not the most efficient way to build the hash map.

Reply:
I'd be tempted to create a view like `changelog_deduped` or write a raw SQL query in diesel something like this...
```
CREATE VIEW latest_asset_log AS
  SELECT al.id,
    al.asset_id,
    al.user_id,
    al.status,
    al.comment,
    al.type,
    al.reason,
    al.log_datetime
  FROM (
    SELECT asset_id, MAX(log_datetime) AS latest_log_datetime
    FROM asset_log
    GROUP BY asset_id
  ) grouped
  INNER JOIN asset_log al
    ON al.asset_id = grouped.asset_id AND al.log_datetime = grouped.latest_log_datetime;

CREATE INDEX ix_asset_log_asset_id_log_datetime ON asset_log (asset_id, log_datetime);
```

Or just run a query for each ID maybe? Could be a lot but bounded to how many we display in a page (e.g. 20?)

I think reading all each asset's log entry just to get that latest could be a potential performance issue in future if there's 1000s of status logs...

I'm also not 100% if we're guaranteed to get the right result with this implementation. Have you found any docs to suggest that `collect()` will always take the first record for each id as the hashmap entry? I've had a look but can't find anything definitive...